### DISCUSSION TITLE
Plotting basic raman spectra

### DISCUSSION THREAD

**Discussion by sabourizk** [spectrochempy/spectrochempy#689] [Q&A] [answered]
Hi, I'm new to spectrochempy and would love some help plotting my raman data. There are there different lines (defined in my y dimension). I'd like to plot them, create a legend with labels I create (and not the self-generated ones) and axis limits, and exclude any data points that exceed a certain value. Any help would be very much appreciated. I'm not sure why I keep getting stuck. 

I tried using the plot_multiple method, but I want all the lines to be solid (and not dashed or dotted). I tried just using the data.plot() method but I can't figure out how to remove data points above a certain value in my y axis (or my "data" dimension). 

Thanks!

**Comment by fernandezc** [comment]
For plot multiple you can use the new version of this function (not yet released): 
https://www.spectrochempy.fr/latest/reference/generated/spectrochempy.NDDataset.html#spectrochempy.NDDataset.plot_multiple

To use it you need to use conda or mamba and install the dev version

`mamba install -c spectrocat/label/dev spectrochempy  `

If you send me some sample spectra, I can write a small exemple to show you how to do the plotting you want (this could be introduced in our documentation as well). To do advanced plotting, it may be sometimes necessary to use directly the matplotlib library.

**Comment by fernandezc** [comment]
", and exclude any data points that exceed a certain value".

For this you could use masks. 

In a next post I will give you an example of code.

**Reply by fernandezc** [reply]
example of code:
```python
import spectrochempy as scp
A = scp.read_labspec("ramandata/SMC1-Initial_RT.txt")
_ = A.plot(ylim=(0,3500))
A[A.data>2000] = scp.MASKED
_ = A.plot(ylim=(0,3500))
```

Which gives this:
<img width="590" alt="image" src="https://github.com/spectrochempy/spectrochempy/assets/2428474/4ec80b2c-feb0-4dcd-b704-c58deaf63c0c">

**Comment by sabourizk** [comment]
[lasertest1.txt](https://github.com/spectrochempy/spectrochempy/files/11737527/lasertest1.txt)

thank you!! I'll keep trying it out. I attached a sample txt file that I've been trying to work with. Thanks so much!

**Reply by fernandezc** [reply]
Well, I found a bug trying to write a simple example with plot_multiple. 
I will try to correct it ASAP. 
So don't loose time until a new dev version is done.

**Comment by fernandezc** [comment]
OK plot_multiple is now fixed. You can get the last DEV version, and here is a sample code on how you can proceed.
(I did not try with your data, but it is probable similar) 

```python
import spectrochempy as scp

B1 = scp.read("ramandata/serie190214-1.txt")

# Basic plot
_ = B1.plot(cmap=None, lw=1)

# limit the x range
B2 = B1[:, 60.0:] # note the float number to specify that we use coordinates

# detrend
B3 = scp.detrend(B2)
_ = B3.plot(cmap=None)

# denoising
pca = scp.PCA(n_components=10)
pca.fit(B3)
B4 = pca.inverse_transform()
_ = B4.plot(cmap=None)

# take only some spectra (5) 
B5 = B4[:5]
B5.plot(cmap=None)

# plot multiple with offset and labels
prefs = B5.preferences
prefs.figure.figsize = (8,6) 
datasets = list(B5)
_ = scp.plot_multiple(datasets, 
                      method="pen",
                      legend="best",
                      labels=["A", "B", "C", "D", "E"],
                      color = ["black", "red", "green", "blue", "violet"],
                      lw=[1, 2.5, 1, 1, 1], # line width
                      ls='-',  # solid line
                      shift=1000, # vertical shift
                     )
```
B1 plot
<img width="448" alt="image" src="https://github.com/spectrochempy/spectrochempy/assets/2428474/b4ed092e-40fc-4965-9f43-c8a270758bea">
B3 plot
<img width="448" alt="image" src="https://github.com/spectrochempy/spectrochempy/assets/2428474/04a9ce84-2a66-408b-9c7e-71a8a02ace89">
B4 plot
<img width="448" alt="image" src="https://github.com/spectrochempy/spectrochempy/assets/2428474/2ad8bf3e-abe7-4dd6-ac4c-54b69e66a0e2">
plot_multiple
<img width="669" alt="image" src="https://github.com/spectrochempy/spectrochempy/assets/2428474/c56f34db-4ff0-4428-b60b-cf082059a641">

**Reply by sabourizk** [reply]
Thank you!! For some reason, I get this error: | TypeError: spectrochempy.core.dataset.arraymixins.ndplot.NDPlot.plot() got multiple values for keyword argument 'ls' for B5. I tried linestyle = '-' and that worked though! thanks so much!

**Reply by fernandezc** [reply]
I open an issue from this discussion in order to not forget to look at this problem.

**Reply by fernandezc** [reply, marked as answer]
In principle this is now solved : see https://github.com/spectrochempy/spectrochempy/issues/690#issuecomment-1599133122.

I mark this discussion as answered for now.
